seisbench/models/base.py

```diff
@@ -141,60 +143,257 @@ def download_metadata_callback(metadata_path):
                 )
+                with open(metadata_path, "r") as f:
+                    weights_metadata = json.load(f)
 
-                if metadata_path.is_file():
-                    with open(metadata_path, "r") as f:
-                        config = json.load(f)
-                    detail_weights[weight] = config.get("docstring", None)
-                else:
-                    detail_weights[weight] = None
+        return weights_metadata.get("docstring", None)
 
-            weights = detail_weights
+    @staticmethod
+    def _parse_weight_filename(filename):
+        """
+        Parses filename into weight name, file type and version string.
+        Returns None, None, None if file can not be parsed.
+        """
+        if isinstance(filename, Path):
+            filename = filename.name
+
+        name = None
+        version_str = None
+
+        for ftype in ["json", "pt"]:
+            p = filename.find(f".{ftype}")
+            if p != -1:
+                name = filename[:p]
+                remainder = filename[p + len(ftype) + 1 :]
+                if len(remainder) > 0:
+                    if remainder[:2] != ".v":
+                        return None, None, None
+                    version_str = remainder[2:]
+                break
+
+        if name is None and version_str is None:
+            ftype = None
+
+        return name, ftype, version_str
+
+    @classmethod
+    def list_versions(cls, name, remote=True):
+        """
+        Returns list of available versions for a given weight name.
+
+        :param name: Name of the queried weight
+        :type name: str
+        :param remote: If true, reports both locally available versions and versions in the remote repository.
+                       Otherwise only reports local versions.
+        :type remote: bool
+        :return: List of available versions
+        :rtype: list[str]
+        """
+        cls._cleanup_local_repository()
+
+        files = [x.name for x in cls._model_path().iterdir()]
+        versions = cls._get_versions_from_files(name, files)
+
+        if remote:
+            remote_path = cls._remote_path()
+            files = seisbench.util.ls_webdav(remote_path)
+            remote_versions = cls._get_versions_from_files(name, files)
 
-        return weights
+            if "" in remote_versions:
+                remote_versions = [x for x in remote_versions if x != ""]
+                # Need to download config file to check version
+                file_version = cls._get_version_of_remote_without_suffix(name)
+                remote_versions.append(file_version)
+
+            versions = list(set(versions + remote_versions))
+
+        return sorted(versions)
+
+    @classmethod
+    def _get_version_of_remote_without_suffix(cls, name):
+        """
+        Gets the version in the remote config without version suffix. Assumes this config exists.
+        """
```

Comment:
This is slightly pedantic, but just adding in here for reference. The way the method and docstring are worded, calling ```_get_version_of_remote_without_suffix(...)``` is slightly confusing as the file suffix is __.v*__ so, strictly speaking, removing the suffix would be to remove all of this. As we get back the integer component of the version suffix, can we change to ```_get_version_number_only_from_remote``` or similar.

```python 
sbm.PhaseNet._get_version_of_remote_without_suffix('stead')

>>> '1'
```

Reply:
Actually, the function does something slightly different. Assuming there is a file called `[name].json`, it returns the version reported in the json. This is required to figure out if this file might be the newest version. I've updated the docstring to make this more clear.